core/src/sql/statements/define/analyzer.rs

```diff
@@ -50,21 +48,20 @@ impl DefineAnalyzerStatement {
 		}
 		// Process the statement
 		let key = crate::key::database::az::new(opt.ns()?, opt.db()?, &self.name);
-		run.add_ns(opt.ns()?, opt.strict).await?;
-		run.add_db(opt.ns()?, opt.db()?, opt.strict).await?;
-		// Persist the definition
-		run.set(
+		txn.get_or_add_ns(opt.ns()?, opt.strict).await?;
+		txn.get_or_add_db(opt.ns()?, opt.db()?, opt.strict).await?;
+		txn.set(
 			key,
 			DefineAnalyzerStatement {
-				// Don't persist the "IF NOT EXISTS" clause to schema
+				// Don't persist the `IF NOT EXISTS` clause to schema
 				if_not_exists: false,
 				..self.clone()
 			},
 		)
 		.await?;
-		// Release the transaction
-		drop(run); // Do we really need this?
-		   // Ok all good
+		// Clear the cache
+		txn.clear();
```

Comment:
I am wondering if we should not have a selective cache clearing (upcoming PR?)
```
txn.clear_analyzer();
```

Reply:
Yeah we can definitely do this. I thought it was safer (with such a big PR) just to clear everything, especially as this only happens after a `DEFINE` statement is executed.

Also this is currently only a transaction cache. So most `DEFINE` statements are likely to take place within separate transactions anyway. However when we use the cache for more, this will be more important 👍.